lib/collection/proxy-config.js

```diff
@@ -102,176 +88,18 @@ _.assign(ProxyConfig.prototype, /** @lends ProxyConfig.prototype */ {
             return;
         }
         _.has(options, 'server') && this.server.update(options.server);
-        _.isString(options.match) && (this.match = options.match);
+        _.isString(options.match) && (this.match = new UrlMatchPattern(options.match));
```

Comment:
My take, matching a subset of protocol can also be thought as "pattern". What do you think?